game/src/main/scala/io/aigar/controller/AigarStack.scala

```diff
@@ -1,18 +1,31 @@
 package io.aigar.controller
+import io.aigar.controller.response.{ErrorResponse}
 
 import org.scalatra._
 import scalate.ScalateSupport
 
-trait AigarStack extends ScalatraServlet with ScalateSupport {
+import org.json4s.{DefaultFormats, Formats}
+import org.scalatra.json._
 
-  notFound {
-    // remove content type in case it was set through an action
-    contentType = null
-    // Try to render a ScalateTemplate if no route matched
-    findTemplate(requestPath) map { path =>
-      contentType = "text/html"
-      layoutTemplate(path)
-    } orElse serveStaticResource() getOrElse resourceNotFound()
+trait AigarStack extends ScalatraServlet with ScalateSupport with JacksonJsonSupport {
+  protected implicit val jsonFormats: Formats = DefaultFormats
+
+  before() {
+    contentType = formats("json")
+  }
+
+
+  trap(404) {
```

Comment:
Just a thought: do we really care about the error *message* itself? If all we want is to trap `>= 400` HTTP status codes to wrap them in our custom JSON format, we could just [trap a range of statuses](http://www.scalatra.org/2.3/api/index.html#org.scalatra.ScalatraServlet@trap(codes:Range)(block:=>Any):Unit) and avoid defining all the ones that we use, e.g.:
```
trap(400 to 599) {
   ...
}
```
I guess that could fit with our robustness project goal (by ensuring that we don't forget one and that the errors will always have the same JSON format)?

The logic of `returnError` could then move directly to the `trap`.